mediatr/mediator.py

```diff
@@ -101,32 +108,39 @@ def send(self,request):
         
         """
         raise_if_request_invalid(request)
-        handler  = __handlers__[request.__class__] if __handlers__.get(request.__class__) else None
-        raise_if_handler_not_found(handler,__handlers__)
+        handler = __handlers__[request.__class__] if __handlers__.get(request.__class__) else None
+        raise_if_handler_not_found(handler, __handlers__)
         handler_func = None
-        handler_obj=None
+        handler_obj = None
         if inspect.isfunction(handler):
             handler_func = handler
         else:
             handler_obj = self.handler_class_manager(handler)
             handler_func = handler_obj.handle
         behaviors = find_behaviors(request)
         gen = None
+
         def start_func():
             for behavior in behaviors:
                 if inspect.isfunction(behavior):
                     beh_func = behavior
                 else:
-                    beh_obj = self.handler_class_manager(behavior,True)
+                    beh_obj = self.handler_class_manager(behavior, True)
                     beh_func = beh_obj.handle
-                yield beh_func(request,next_func)
+                yield beh_func(request, next_func)
             yield handler_func(request)
+
         gen = start_func()
-        async def next_func():
-            return await __return_await__(next(gen))
-        return __get_result_block__(next_func())
+        if inspect.iscoroutinefunction(self):
+            async def next_func():
+                return await __return_await__(next(gen))
+
+            return __get_result_block__(next_func())
+        else:
+            def next_func():
```

Comment:
Fallback to get everything running within sync processes e.g. django with ORM

Reply:
Thank you for edit code. Will edits work in python 3.5 ? I see that version changed

Reply:
Why don't you initialize the instance of mediator ? Mediator().send().